pkg/whitesource/whitesource.go

```diff
@@ -115,24 +126,42 @@ func (s *System) GetProjectsMetaInfo(productToken string) ([]Project, error) {
 }
 
 //GetProjectToken returns the project token for a project with a given name
-func (s *System) GetProjectToken(productToken, projectName string) (string, error) {
+func (s *System) GetProjectByName(productToken, projectName string) (*Project, error) {
 	projects, err := s.GetProjectsMetaInfo(productToken)
 	if err != nil {
-		return "", errors.Wrap(err, "failed to retrieve WhiteSource project meta info")
+		return nil, errors.Wrap(err, "failed to retrieve WhiteSource project meta info")
 	}
 
 	for _, project := range projects {
 		if projectName == project.Name {
-			return project.Token, nil
+			return &project, nil
 		}
 	}
-	return "", nil
+	return nil, errors.New(fmt.Sprintf("Failed to find a project with name: %s", projectName))
+}
+
+// get all projects tokens for a list of given project ids
+func (s *System) GetProjectsByIds(productToken string, projectIds []int64) ([]Project, error) {
+	projectsMatched := []Project{}
+
+	projects, err := s.GetProjectsMetaInfo(productToken)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to retrieve WhiteSource project meta info")
+	}
+
+	for _, project := range projects {
+		for _, projectId := range projectIds {
+			if projectId == project.ID {
+				projectsMatched = append(projectsMatched, project)
+			}
+		}
+	}
+	return projectsMatched, nil
 }
 
 //GetProjectTokens returns the project tokens for a list of given project names
-func (s *System) GetProjectTokens(productToken string, projectNames []string) ([]string, error) {
+func (s *System) GetProjectTokensByNames(productToken string, projectNames []string) ([]string, error) {
```

Comment:
Where did the function `GetProjectTokens()` go?
Seems like an incompatible change. 
We should stay compatible here if possible. If we want to have better names, let's at least keep the original names and redirect the old one to the function with the new name.